website/notifications/utils.py

```diff
@@ -188,72 +257,91 @@ def format_data(user, node_ids):
     return items
 
 
-def format_user_subscriptions(user, data):
+def format_user_subscriptions(user):
     """ Format user-level subscriptions (e.g. comment replies across the OSF) for user settings page"""
-    user_subscriptions = [s for s in model.NotificationSubscription.find(Q('owner', 'eq', user))]
-    for subscription in constants.USER_SUBSCRIPTIONS_AVAILABLE:
-        event = serialize_event(user, subscription, constants.USER_SUBSCRIPTIONS_AVAILABLE, user_subscriptions)
-        data.append(event)
-
-    return data
-
-
-def serialize_event(user, subscription, subscriptions_available, user_subscriptions, node=None):
+    user_subs_available = list(constants.USER_SUBSCRIPTIONS_AVAILABLE.keys())
+    subscriptions = [
+        serialize_event(user, subscription,
+                        event_description=user_subs_available.pop(user_subs_available.index(getattr(subscription, 'event_name'))))
+        for subscription in get_all_user_subscriptions(user)
+        if getattr(subscription, 'event_name') in user_subs_available
```

Comment:
These tend to be of one more depth, most of the ```getattr``` are on either side of 'in' and therefore are getting lists or elements that might be contained by the right-hand side or the left-hand operator is looking to see its membership in the list returned from ```getattr```. ```hasattr``` would be true in most of these cases as they should have event_name.